sdk/core/core-client-rest/review/core-client.api.md

```diff
@@ -69,6 +57,16 @@ export type HttpResponse = {
 // @public
 export function isCertificateCredential(credential: unknown): credential is CertificateCredential;
 
+// @public
+export type PathParameters<TRoute extends string> = TRoute extends `${infer _Head}/{${infer _Param}}${infer Tail}` ? [
+pathParameter: string,
+...pathParameters: PathParameters<Tail>
+] : [
+];
+
+// @public
+export type PathUnchecked<MethodType extends "Full" | "Partial" = "Full"> = <TPath extends string>(path: TPath, ...args: PathParameters<TPath>) => MethodType extends "Full" ? ResourceMethods : Partial<ResourceMethods>;
```

Comment:
This trick with the generic parameter is pretty clever! 

I'm curious though, when does "Full" get used? And does this mean by default someone would have to feature detect if delete/get/head/etc were available before calling them?

Reply:
Would it be sufficient to take either ResourceMethods or Partial<ResourceMethods> instead of "Full" or "Partial"? Not sure the magic pays for itself personally.

Reply:
discussed offline with @bterlson and adding these generics doesn't provide much value for the complexity and maintainability burden. We'll keep the type as it was  `path: Function` and  document how it gets overridden in the generated code